test/e2e/metrics_test.go

```diff
@@ -89,134 +129,265 @@ func NewMetricsTestConfig(t *testing.T, client, selector, clusterRole, clusterBi
+		Tty:    false,
+	}
+
+	err = executor.StreamWithContext(ctx, streamOpts)
+	require.NoError(c.t, err, "Error streaming exec request: %v", stderr.String())
 
-	c.t.Log("Validating the metrics endpoint")
-	curlCmd := exec.Command(c.client, "exec", c.curlPodName, "-n", c.namespace, "--",
-		"curl", "-v", "-k", "-H", "Authorization: Bearer "+token, c.metricsURL)
-	output, err := curlCmd.CombinedOutput()
-	require.NoError(c.t, err, "Error calling metrics endpoint: %s", string(output))
-	require.Contains(c.t, string(output), "200 OK", "Metrics endpoint did not return 200 OK")
+	// Combine stdout + stderr
+	combined := stdout.String() + stderr.String()
+	require.Contains(c.t, combined, "200 OK", "Metrics endpoint did not return 200 OK")
 }
 
-// cleanup removes the created resources. Uses a context with timeout to prevent hangs.
-func (c *MetricsTestConfig) cleanup() {
+// cleanup deletes the test resources
+func (c *MetricsTestConfig) cleanup(ctx context.Context) {
 	c.t.Log("Cleaning up resources")
-	_ = exec.Command(c.client, "delete", "clusterrolebinding", c.clusterBinding, "--ignore-not-found=true", "--force").Run()
-	_ = exec.Command(c.client, "delete", "pod", c.curlPodName, "-n", c.namespace, "--ignore-not-found=true", "--force").Run()
+	policy := metav1.DeletePropagationForeground
 
-	// Create a context with a 60-second timeout.
-	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
-	defer cancel()
+	// Delete the ClusterRoleBinding
+	_ = c.kubeClient.RbacV1().ClusterRoleBindings().Delete(ctx, c.clusterBinding, metav1.DeleteOptions{
+		PropagationPolicy: &policy,
+	})
+	waitForClusterRoleBindingDeletion(ctx, c.t, c.kubeClient, c.clusterBinding)
 
-	// Wait for the ClusterRoleBinding to be deleted.
-	if err := waitForDeletion(ctx, c.client, "clusterrolebinding", c.clusterBinding); err != nil {
-		c.t.Logf("Error waiting for clusterrolebinding deletion: %v", err)
-	} else {
-		c.t.Log("ClusterRoleBinding deleted")
-	}
+	// "Force" delete the Pod by setting grace period to 0
+	gracePeriod := int64(0)
+	_ = c.kubeClient.CoreV1().Pods(c.namespace).Delete(ctx, c.curlPodName, metav1.DeleteOptions{
+		GracePeriodSeconds: &gracePeriod,
+		PropagationPolicy:  &policy,
+	})
+	waitForPodDeletion(ctx, c.t, c.kubeClient, c.namespace, c.curlPodName)
+}
 
-	// Wait for the Pod to be deleted.
-	if err := waitForDeletion(ctx, c.client, "pod", c.curlPodName, "-n", c.namespace); err != nil {
-		c.t.Logf("Error waiting for pod deletion: %v", err)
+// waitForClusterRoleBindingDeletion polls until the named ClusterRoleBinding no longer exists
+func waitForClusterRoleBindingDeletion(ctx context.Context, t *testing.T, kubeClient kubernetes.Interface, name string) {
+	err := wait.PollUntilContextTimeout(ctx, 2*time.Second, 60*time.Second, false, func(ctx context.Context) (bool, error) {
+		_, err := kubeClient.RbacV1().ClusterRoleBindings().Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			if strings.Contains(err.Error(), "not found") {
```

Comment:
nit: you can use `k8s.io/apimachinery/pkg/api/errors:IsNotFound(err)`. Here and potentially other places

Reply:
done in two places in next commit